frontend/public/components/nav/section.tsx

```diff
@@ -185,10 +193,18 @@ export const NavSection = connect(navSectionStateToProps)(
         getChildren() {
           const { id, title, children, activePerspective: perspective } = this.props;
           const Children = React.Children.map(children, this.mapChild) || [];
-          const childItems = sortExtensionItems(this.getNavItemExtensions(perspective, title, id));
```

Comment:
I'd suggest these changes:

```diff
 export const NavSection = connect(navSectionStateToProps)(
   withExtensions<NavSectionExtensionProps>({
-    navItemExtensions: isNavItem,
+    navItemExtensions: isNavItemOrSeparator,
     perspectiveExtensions: isPerspective,
   })(
```

```diff
           childItems.forEach((item) => {
-            const pluginChild = this.mapChild(createLink(item));
+            const pluginChild = this.mapChild(
+              isSeparator(item) ? (
+                <SeparatorComponent name={`separator-${item.properties.id}`} />
+              ) : (
+                createLink(item)
+              ),
+            );
```

```diff
 type NavSectionExtensionProps = {
-  navItemExtensions: LoadedExtension<NavItem>[];
+  navItemExtensions: LoadedExtension<NavItem | Separator>[];
   perspectiveExtensions: Perspective[];
 };
```

Reply:
Also, the `title` parameter of `getNavItemExtensions` method seems to be unused, I'd suggest removing it.

Reply:
removed the title prop,
yes, the rename is nicer, now it's clear it'e the component that render the sepertor.